File: cs50_week2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include<math.h>
#include <ctype.h>
int main(void)
{
    string input = get_string("Text: ");

    float letters = 0;
    float words = 0;
    float sentences = 0;
    for (int i = 0; i < strlen(input); i++)
    {
        if (isalpha(input[i]))
        {
            letters++;
        }
        if (isspace(input[i]) || i == strlen(input) - 1)
        {
            words++;
        }
        if (input[i] == '!' || input[i] == '?' || input[i] == '.')
        {
            sentences++;
        }
    }
    //lettters per 100 words
    float lettersPer = letters / words * 100.0;
    //sentences per 100 words
    float sentencesPer = sentences / words * 100.0;
    float grade = 0.0588 * lettersPer - 0.296 * sentencesPer - 15.8;
    int rounded = round(grade);
    if (rounded > 16)
    {
        printf("Grade 16+\n");
    }
    else if (rounded < 0)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", rounded);
    }

}

File: cs50_week6/sentimental-readability/readability.py
text = input("Text: ")

letters = 0
words = 1
sentences = 0
for element in text:
    if element.isalpha() == True:
        letters += 1
    if element == " ":
        words += 1
    if element == "!" or element == "?" or element == ".":
        sentences += 1

lettersPer100Words = letters / words * 100

sentencesPer100Words = sentences / words * 100

grade = round(0.0588 * lettersPer100Words - 0.296 * sentencesPer100Words - 15.8)

if grade < 1:
    print("Before Grade 1")
elif grade > 16:
    print("Grade 16+")
else:
    print(f"Grade {grade}")